libbeat/publisher/beat/pipeline.go

```diff
@@ -57,6 +65,29 @@ type ClientConfig struct {
 	ACKLastEvent func(Event)
 }
 
+// ClientEventer provides access to internal client events.
+type ClientEventer interface {
+	Closing() // Closing indicates the client is being shutdown next
```

Comment:
why a bool? These are notifications a beat can register too (e.g. adapt wgEvents in filebeat). The listener can not influence the publisher pipeline. It's more a Listener/Observer pattern, no Delegate or such.